R/predict.R

```diff
@@ -206,19 +220,36 @@ predict_obs_cases <- function(
     min_date = min_date,
     max_date = max_date,
     horizon = horizon,
+    day_of_week = day_of_week,
     call = call
   )
 
-  # Use `posterior_samples()` over `fitted_samples()` to get response
-  # w/ obs uncertainty
-  fitted <- gratia::posterior_samples(
-    object[["model"]],
+  args <- list(
+    model = object[["model"]],
     data = newdata,
-    unconditional = TRUE,
     n = n,
     seed = seed,
+    unconditional = TRUE,
+    newdata.guaranteed = TRUE,
     ...
   )
+  # Turn off day of week only if requested and also
+  # was set to on in the model fitting
+  if (
+    is.factor(object[["data"]][["day_of_week"]]) && rlang::is_false(day_of_week)
+  ) {
+    args[["exclude"]] <- "s(day_of_week)"
+  }
+  # Suppress mgcv::gam warning that factor levels are far from the data
+  # "factor levels 1 not in original fit". This warning is spurious bc
+  # we're explicitiy excluding the day-of-week random effect smooth but
+  # need to have some placeholder value in the dataset
+  fitted <- suppressWarnings(do.call(
```

Comment:
I think this is fine but it might be nice to have an issue to just catch this specific warning vs all of them - I can imagine CRAN not being happy with this catch all approach.

Reply:
https://github.com/CDCgov/cfa-gam-rt/pull/89/commits/05d81040918e917351e9e04ae616d38360b2973d